### PR TITLE
Adds new props to contacts

### PR DESCRIPTION
**Pull request by NicDominguez**
Adds priority, mode, and dayOfYear props to contact for use in GRM-Dashboard

### CODE REVIEW

File: src/types/alert.ts [hunk 1]
```diff
@@ -25,16 +25,18 @@ export type Alert = {
...
-  timestamp?: number;
-  selected?: boolean;
-  new?: boolean;
-  expanded?: boolean;
-  acknowledged?: boolean;
-  contactRefId: string;
-};
+export type ModifyAlertParams = AtLeast<Alert, 'id' | 'contactRefId'>;
+
+// export type ModifyAlertParams = {
```
Comment (brikiernan) [inline src/types/alert.ts]: Lets remove the old commented out type if not needed anymore.

File: src/types/contacts.ts [hunk 2]
```diff
@@ -30,33 +39,39 @@ export type Contact = {
...
-  endTimestamp?: number;
-  resolution?: string;
-  resolutionStatus?: string;
-  selected?: boolean;
-  alerts?: Alert[];
-  mnemonics?: Mnemonic[];
-};
+export type ModifyContactParams = AtLeast<Contact, 'id'>;
+
+// export type ModifyContactParams = {
```
Comment (brikiernan) [inline src/types/contacts.ts]: Lets remove the old commented out type if not needed anymore.

File: src/types/mnemonic.ts [hunk 3]
```diff
@@ -25,16 +25,18 @@ export type Mnemonic = {
...
-  thresholdMin?: number;
-  currentValue?: number;
-  subsystem?: string;
-  childSubSystem?: string;
-  measurement?: string;
-  contactRefId: string;
-};
+export type ModifyMnemonicParams = AtLeast<Mnemonic, 'id' | 'contactRefId'>;
+
+// export type ModifyMnemonicParams = {
```
Comment (brikiernan) [inline src/types/mnemonic.ts]: Lets remove the old commented out type if not needed anymore.

File: src/types/util.ts [hunk 4]
```diff
@@ -53,3 +53,9 @@ export type StructuredData<T> = {
   dataById: { [key: string]: T };
   dataIds: string[];
 };
+
+export type Priority = 'Low' | 'Medium' | 'High';
+
+export type Mode = 'Full Automation' | 'Semi-Automation' | 'Manual';
+
+export type AtLeast<T, K extends keyof T> = Partial<T> & Pick<T, K>;
```
Comment (brikiernan) [inline src/types/util.ts:61]: This AtLease type is awesome Nic, nice work!

File: src/utils/index.ts [hunk 5]
```diff
@@ -49,3 +49,12 @@ export const randomMinutes = (min: number, max: number) => {
 export const randomSeconds = (min: number, max: number) => {
   return 1000 * between({ min, max });
 };
+
+export const getDayOfYear = (date: Date) => {
+  const now = new Date(date);
```
Comment (brikiernan) [inline src/utils/index.ts]: If you have to pass in a type of Date, is line 54 needed? I could see it if you could pass in a type of Date, or string, or number. Unless I'm missing something.

**Review by brikiernan — CHANGES_REQUESTED** [review]
@NicDominguez I have a few changes requested, let me know if I'm off base on something. Thanks.

File: src/types/util.ts (hunk 4 above)
Comment (NicDominguez) [reply src/types/util.ts:61]: Okay, good. Wanted to make sure you liked it before removing the comments.

File: src/utils/index.ts (hunk 5 above)
Comment (NicDominguez) [reply src/utils/index.ts]: Good catch, it's not necessary.

**Review by brikiernan — APPROVED** [review]
👍

**Pull request merged**